[cscms03] iP

cscms03 opened the pull request:
# Minnim

>“Your mind is for having ideas, not holding them.” – David Allen 

Minnim frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://github.com/cscms03/ip.git).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) throws MinnimMissingTaskDetailException, MinnimMissingDateException,
            MinnimTargetTaskNumNotFoundException, MinnimNoTaskFoundException {
    new Minnim("data/minnim.Minnim.txt").run();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

cscms03 committed:
Rename class Duke -> Minnim

cscms03 committed:
Add greet and exit feature

cscms03 committed:
Add echo feature

cscms03 committed:
Add Adding and List feature

cscms03 committed:
Add Task class

cscms03 committed:
Modify Task class

cscms03 committed:
Add 'Mark as Done' feature

cscms03 committed:
Add inherited classes of Task class

cscms03 committed:
Implemented getDescription and constructors

cscms03 committed:
Add Todo, Deadline, Event feature

cscms03 committed:
Fix minor bug with space

cscms03 committed:
Add UI Testing inputs and expected outputs

cscms03 committed:
Add Exception classes and support for them

cscms03 committed:
Add Exception classes and support for them

cscms03 committed:
Add exception handling feature

cscms03 committed:
Modify tasks array to ArrayList for implementing Delete feature

cscms03 committed:
Add Delete feature

cscms03 committed:
Update test cases for new Delete feature

cscms03 committed:
Add TargetTaskNumNotFoundException class for handling error when target task number is not given

cscms03 committed:
Add support to the TargetTaskNotFoundException class

cscms03 committed:
Add enum for refactoring

cscms03 committed:
Refactor Minnim class using ENUM and switch

cscms03 committed:
Add toFIleString method to Task class and extending classes

cscms03 committed:
Add "save list to local file" feature to local file

cscms03 committed:
Revert "Add "save list to local file" feature to local file"

This reverts commit 6a78e3b17cf6abb23a22fe863494274d25783cf8.

cscms03 committed:
Reapply "Add "save list to local file" feature to local file"

This reverts commit d19e55a700a919ac91a22416291eb20e5ede497d.

cscms03 committed:
Edit comment

cscms03 committed:
Merge branch 'branch-Level-7'

cscms03 committed:
Add date parse feature

cscms03 committed:
Replace wildcard import with specific classes

cscms03 committed:
Merge branch 'branch-Level-8'

cscms03 committed:
Refactor to comply with OOP

cscms03 committed:
Update exception handling

cscms03 committed:
Merge branch 'branch-MoreOOP'

cscms03 committed:
Organize classes into packages and refactor

cscms03 committed:
Merge branch 'branch-A-Packages'

cscms03 committed:
Merge branch 'add-gradle-support'

cscms03 committed:
Refactor code from moving Minnim.java file into package

cscms03 committed:
Edit build.gradle to make Gradle build, run, test, etc

cscms03 committed:
Fix bug when loading existing tasks

cscms03 committed:
Add JUnit testcases for Storage class

cscms03 committed:
Add JUnit testcases for other classes such as Task, TaskList

cscms03 committed:
Merge branch 'branch-A-JUnit'

cscms03 committed:
Add JavaDoc for Parser class and methods

cscms03 committed:
Add JavaDoc for Storage class and methods

cscms03 committed:
Add JavaDoc for TaskList class and methods

cscms03 committed:
Add JavaDoc for Task and its child classes and methods

cscms03 committed:
Add JavaDoc for MinnimException class and some child classes

cscms03 committed:
Implement Find feature with exception handling

cscms03 committed:
Add JavaDoc for Minnim class

cscms03 committed:
Organize import order complying to CodingStandard

cscms03 committed:
Merge branch 'branch-A-CodingStandard'

cscms03 committed:
Merge branch 'branch-Level-9'

cscms03 committed:
Merge branch 'branch-A-JavaDoc'

MrAsparag00se commented on src/main/java/Task.java:
```diff
@@ -23,4 +23,9 @@ public void setUnmarked() {
     }
+
+    public String toFileString() {
+        return this.getClass().getSimpleName() + " | " +
```
What is a simple name?

MrAsparag00se commented on src/main/java/Minnim.java:
```diff
@@ -1,10 +1,67 @@
+
+                if (task != null) {
+                    if (isDone) {
+                        task.setMarked();
```
Is this too much indentation?

MrAsparag00se commented on src/main/java/Minnim.java:
```diff
@@ -7,7 +7,7 @@
     private Scanner s = new Scanner(System.in);
     private ArrayList<Task> tasks = new ArrayList<>();
-    private final String dateDivider = "/";
+    private final String DATE_DIVIDER = "/";
```
Naming this a divider may be confused with being a divisor when you probably mean the parsing delimiter

MrAsparag00se commented on src/main/java/Deadline.java:
```diff
@@ -1,19 +1,38 @@
+            return LocalDate.parse(deadlineStr, formatter);
+        } catch (DateTimeParseException e) {
+            // Handle invalid date format here
+            System.out.println("Invalid date format. Please use yyyy-MM-dd.");
```
What about invalid times since this seems to only validate date formats and not time 😐

MrAsparag00se commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,73 @@
+
+                if (task != null) {
+                    if (isDone) {
+                        task.setMarked();
```
Perhaps too many layers of indentation? Consider refactoring :)

MrAsparag00se reviewed (commented):
Just some minor changes, but overall code looks neat and readable. Keep it up :)

jingchun19 commented on src/main/java/minnim/parser/Parser.java:
```diff
@@ -0,0 +1,79 @@
+                System.exit(0);
+            default:
+                ui.showUnknownCommandMessage();
+        }
```
Good use of toLowerCase for ease of use

jingchun19 commented on src/main/java/minnim/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+                    task = new Deadline(description, parts[3]);
+                } else if (type.equals("Event")) {
+                    task = new Events(description, parts[3], parts[4]);
+                }
```
perhaps you could use toLowerCase here as well?

jingchun19 commented on src/main/java/minnim/task/TaskList.java:
Great work on the java docs

jingchun19 commented on src/main/java/minnim/ui/Ui.java:
I didn't think of using the class UI like this. Great work

jingchun19 commented on src/test/java/minnim/task/TaskListTest.java:
Very detailed testing.

jingchun19 reviewed (commented):
Overall, very neat code base, especially the exception folder. Keep up the good work!

cscms03 committed:
Update comments about date and time

cscms03 committed:
Update gradle file for setting up JavaFX

cscms03 committed:
Update parsing feature from console output to String return value

cscms03 committed:
Add Basic GUI to Minnim

cscms03 committed:
Merge branch 'branch-Level-10'

cscms03 committed:
Add assertions to classes

cscms03 committed:
Refactor methods to adhere SLAP

There are several long methods that do not adhere to Single Level of
Abstraction Principle. Refactoring these methods improves code quality.

cscms03 committed:
Delete unnecessary file and lines

cscms03 committed:
Replace to String.format() to improve readability

cscms03 committed:
Add and update JavaDoc for public methods

cscms03 committed:
Merge pull request #2 from cscms03/branch-A-Assertions

Add assertions to classes

cscms03 committed:
Merge branch 'master' into branch-A-CodeQuality

cscms03 committed:
Merge pull request #3 from cscms03/branch-A-CodeQuality

Refactor code for better readability, maintainability, and adherence to the Single Responsibility Principle (SLAP).

Simplify methods, remove redundant code, and improve string formatting. 

Update JavaDoc for public methods to enhance documentation clarity.

cscms03 committed:
Merge branch 'master' of https://github.com/cscms03/ip

* 'master' of https://github.com/cscms03/ip:
  Add and update JavaDoc for public methods
  Replace to String.format() to improve readability
  Delete unnecessary file and lines
  Refactor methods to adhere SLAP
  Add assertions to classes

cscms03 committed:
Revert "Merge branch 'master' of https://github.com/cscms03/ip"

This reverts commit ec6dd564316861287022906f934a847cf06304dc, reversing
changes made to 5967dc7da4cb2821dcaea8145c469e7af9a2fd24.

cscms03 committed:
Add UndoStorage class
Stores recent tasks done, or deleted task if applicable.

cscms03 committed:
Update Parser and TaskList class for Undo feature
Add methods such as handleDelete to appropriately process 'undo' command

cscms03 committed:
Add assertions to classes

cscms03 committed:
Remove unncessary file

cscms03 committed:
Refactor find method and update Javadoc to return String value

cscms03 committed:
Refactor readFile method

cscms03 committed:
Refactor JavaDoc for parseCommand method

cscms03 committed:
Remove unnecessary imports and comments

cscms03 committed:
Add and update JavaDoc for Storage, DialogueBox, MainWindow classes

cscms03 committed:
Merge pull request #4 from cscms03/branch-A-Assertions

This merge introduces assertions throughout the codebase to improve 
reliability and catch potential issues early.

cscms03 committed:
Merge branch 'master' into branch-A-CodeQuality

cscms03 committed:
Merge pull request #5 from cscms03/branch-A-CodeQuality

Merge branch-A-CodeQuality to master

This merge incorporates several improvements aimed at enhancing
code quality. Key changes include:

Updates and additions of Javadoc comments for better code
documentation.
Refactoring of methods in the Storage and TaskList classes to
adhere to the Single Level of Abstraction Principle (SLAP).
Removal of unnecessary files and imports to streamline the
codebase.
These enhancements contribute to improved readability, maintainability,
and overall code clarity.

cscms03 committed:
Merge branch 'branch-C-Undo'

The current implementation lacks an effective undo feature, which
limits user flexibility and complicates task management.

Implement the undo functionality to allow users to easily revert
recent actions, enhancing the overall user experience.

This approach is chosen to ensure a seamless interaction with the
application, making it intuitive and responsive to user needs.

cscms03 committed:
Update Event and Deadline Exception handling

Update so that date parsers throw DateTimeParseException with error
messages.

cscms03 committed:
Update TaskList class to show error messages as response

Minnim only shows error message through console, which is not visible to
the users. Users are not notified what type of errors caused.

cscms03 committed:
Merge branch 'branch-A-MoreErrorHandling'

Update handling methods to not simply throw new exceptions, but actually
return String error messages to be shown through UI.

cscms03 committed:
Update JUnit testcases to test String return values

cscms03 committed:
Merge branch 'branch-A-MoreTesting'

Update JUnit testcases to handle String return values.

cscms03 committed:
Add different image for user

cscms03 committed:
Adjust size of chatbot and user profile image

cscms03 committed:
Merge branch 'branch-A-BetterGui'

Minor update to GUI of Minnim chatbot for better visibility

cscms03 committed:
Set window title to product name

cscms03 committed:
Update ReadMe user guide

cscms03 committed:
Fix hard-coded file paths

cscms03 committed:
Remove unused methods in Minnim class

cscms03 committed:
Remove unnecessary System.out.println() calls

cscms03 committed:
Edit build.gradle file to support other OS
